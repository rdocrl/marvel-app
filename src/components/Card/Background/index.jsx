import './Background.scss';

const Background = ({ image, children }) => {
  return (
    <>
      <div
        className="background"
        style={{
          backgroundImage: `url('${image}')`
        }}>
        <div className="background__opacity"></div>
      </div>
      <div className="content">{children}</div>
    </>
  );
};

export default Background;
