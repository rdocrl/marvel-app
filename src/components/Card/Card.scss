.card {
  cursor: pointer;
  width: 206px;

  &__thumbnail {
    transition: border-radius 300ms;

    &:hover {
      border-radius: 12px;
    }

    &--no-thumbnail {
      width: 216px;
      height: 316px;
      display: flex;
      align-items: center;
      justify-content: center;
      background-image: url('../../assets/no-thumbnail.jpg');
    }
  }

  &__title {
    text-align: left;
  }
}
