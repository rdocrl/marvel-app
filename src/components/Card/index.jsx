import './Card.scss';

const Card = ({ title, thumbnail, onClick, isFavorite, id, onToggleFavorite, type }) => {
  return (
    <article
      className="card"
      onClick={onClick}
      title={typeof onClick === 'function' ? 'See details' : ''}>
      {typeof thumbnail === 'string' ? (
        <img src={thumbnail} className="card__thumbnail" alt={title} />
      ) : (
        <div className="card__thumbnail--no-thumbnail">Image not available</div>
      )}
      <div
        className={`card__favorite ${isFavorite ? '' : 'card__favorite--no-favorite'}`}
        title={isFavorite ? 'Remove favorite' : 'Add favorite'}
        onClick={(e) => {
          e.stopPropagation();
          const newFavorite = { id, title, thumbnail, type };
          onToggleFavorite(newFavorite);
        }}>
        <span className="card__star">‚≠ê</span>
      </div>
      <h3 className="card__title">{title}</h3>
    </article>
  );
};

export default Card;
