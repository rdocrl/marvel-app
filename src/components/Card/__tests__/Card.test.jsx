import { render } from '@testing-library/react';
import Card from '../';

describe('<Card /> tests', () => {
  describe('Snapshots', () => {
    it('Should render with no props', () => {
      const { asFragment } = render(<Card />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('Should render with props', () => {
      const { asFragment } = render(<Card title="Card title" thumbnail="" onClick={() => {}} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
