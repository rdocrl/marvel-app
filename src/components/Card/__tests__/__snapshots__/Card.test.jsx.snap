// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> tests Snapshots Should render with no props 1`] = `
<DocumentFragment>
  <article
    class="card"
    title=""
  >
    <div
      class="card__thumbnail--no-thumbnail"
    >
      Image not available
    </div>
    <div
      class="card__favorite card__favorite--no-favorite"
      title="Add favorite"
    >
      <span
        class="card__star"
      >
        ⭐
      </span>
    </div>
    <h3
      class="card__title"
    />
  </article>
</DocumentFragment>
`;

exports[`<Card /> tests Snapshots Should render with props 1`] = `
<DocumentFragment>
  <article
    class="card"
    title="See details"
  >
    <img
      alt="Card title"
      class="card__thumbnail"
      src=""
    />
    <div
      class="card__favorite card__favorite--no-favorite"
      title="Add favorite"
    >
      <span
        class="card__star"
      >
        ⭐
      </span>
    </div>
    <h3
      class="card__title"
    >
      Card title
    </h3>
  </article>
</DocumentFragment>
`;
